multichain-aggregator/multichain-aggregator-proto/proto/v1/multichain-aggregator.proto

```diff
@@ -0,0 +1,75 @@
+syntax = "proto3";
+
+package blockscout.multichainAggregator.v1;
+
+option go_package = "github.com/blockscout/blockscout-rs/multichain-aggregator";
+
+service MultichainAggregatorService {
+  rpc BatchImport(BatchImportRequest) returns (BatchImportResponse) {}
+  rpc QuickSearch(QuickSearchRequest) returns (QuickSearchResponse) {}
+}
+
+message Address {
+  string hash = 1;
+  optional string ens_name = 2;
+  optional string contract_name = 3;
+  optional string token_name = 4;
+
+  enum TokenType {
+    TOKEN_TYPE_UNSPECIFIED = 0;
+    TOKEN_TYPE_ERC_20 = 1;
+    TOKEN_TYPE_ERC_721 = 2;
+    TOKEN_TYPE_ERC_1155 = 3;
+    TOKEN_TYPE_ERC_404 = 4;
+  }
+
+  TokenType token_type = 5;
+  optional bool is_contract = 6;
+  optional bool is_verified_contract = 7;
+  optional bool is_token = 8;
+}
+
+message BlockRange {
+  uint64 min_block_number = 1;
+  uint64 max_block_number = 2;
+}
+
+message Hash {
+  string hash = 1;
+
+  enum HashType {
+    HASH_TYPE_BLOCK = 0;
+    HASH_TYPE_TRANSACTION = 1;
+  }
+
+  HashType hash_type = 2;
+}
+
+message BatchImportRequest {
+  string chain_id = 1;
+  repeated Address addresses = 2;
+  repeated BlockRange block_ranges = 3;
+  repeated Hash hashes = 4;
+  string api_key = 5;
+}
+
+message BatchImportResponse { string status = 1; }
+
+message QuickSearchRequest { string q = 1; }
+
+message QuickSearchResponse {
+  message ChainBlockNumber {
+    int64 chain_id = 1;
+    uint64 block_number = 2;
+  }
+
+  message ChainSearchResult {
+    string explorer_url = 1;
+    repeated Address addresses = 2;
+    repeated Hash blocks = 3;
+    repeated Hash transactions = 4;
+    repeated ChainBlockNumber block_numbers = 5;
+  }
+
+  map<string, ChainSearchResult> items = 1;
+}
```

Comment:
what about pagination?

Reply:
It is a quick search, so pagination is not supported in this case

Reply:
Extended search for particular entities will include pagination, but plan to add it later